Platform/Teams/Tools and Improvements/[2019-06-11] State of E2E Testing on VA.gov.md

```diff
@@ -0,0 +1,53 @@
+
+Some popular non-Selenium alternatives do not yet support full cross-browser functionality, namely for IE. In the case of Puppeteer, Firefox support is already in an experimental state, but there are no plans for IE. There is a third-party package that appears to add IE support, but it might not be dependable for long-term support. Cypress only currently supports Chrome but [has cross-browser support on its roadmap](https://github.com/cypress-io/cypress/issues/310). TestCafe already [supports every browser out of the box](https://devexpress.github.io/testcafe/documentation/using-testcafe/common-concepts/browsers/browser-support.html#officially-supported-browsers) and [has plugins for integrating with Sauce Labs and BrowserStack](https://devexpress.github.io/testcafe/documentation/using-testcafe/common-concepts/browsers/browser-support.html#browsers-in-cloud-testing-services).
+
+### Test Schedule
+How often should we run the tests? Once the tests integrate the API, it might add too much overhead to run the tests on each commit in a PR. Maybe consider running once a day, perhaps before deploying. Maybe deploy to a dedicated test instance.
+
+### Visual Regression
+Both Selenium and Puppeteer can support visual regression testing, but we already use Puppeteer for that purpose. Cypress supports it as well, but also offers video playback and the ability to step backwards (Time Travel) through the test for debugging purposes, which can be incredibly helpful for development. TestCafe can capture screenshots and video as well.
+
+### Accessibility
+We currently do axe accessibility checks in our Nightwatch tests. It has not yet been integrated into the form filler utility, but there is a package (`axe-puppeteer`) that allows us to use axe with Puppeteer. There are similar packages for Cypress and TestCafe. We also have utilities for such features as keyboard testing that we might need to refactor to work within the framework we end up using.
+
+### Transition / Conversion
+How tedious or time-consuming will it be to convert our Nightwatch tests to Puppeteer or another non-Selenium solution? Or if we don't even continue to use Puppeteer, what is the level of effort involved in converting our current Puppeteer-based utilities to another framework? Furthermore, will that solution allow us to adapt our accessibility testing patterns? Documentation might be a good indicator for the ease of transition. Cypress in particular seems to have fairly comprehensive documentation with many examples, which could make it a much smoother process when handling edge cases or making design choices.
+
+### Code Coverage
+It might be nice to set a bar for code coverage, but we may want to focus first on consolidating and standardizing our E2E testing solution. Note that Cypress provides documentation on implementing instrumentation with Istanbul, which we already use for unit test coverage. https://docs.cypress.io/guides/tooling/code-coverage.html
```

Comment:
```suggestion
It might be nice to set a bar for code coverage, but we may want to focus first on consolidating and standardizing our E2E testing solution. Note that Cypress provides [documentation](https://docs.cypress.io/guides/tooling/code-coverage.html) on implementing instrumentation with Istanbul, which we already use for unit test coverage.
```